classes/SpecificPrice.php

```diff
@@ -188,14 +188,15 @@ public function delete()
         return false;
     }
 
-    public static function getByProductId($id_product, $id_product_attribute = false, $id_cart = false)
+    public static function getByProductId($id_product, $id_product_attribute = false, $id_cart = false, $id_price_rule = false)
```

Comment:
```suggestion
    public static function getByProductId($id_product, $id_product_attribute = false, $id_cart = false, $id_price_rule = null)
```

Reply:
For a better type hint. Never mix boolean and integer. Prefer a null value instead